tools/system_libs.py

```diff
@@ -816,9 +816,11 @@ def __init__(self, **kwargs):
       assert not self.is_tracing
 
   def get_files(self):
-    return [shared.path_from_root('system', 'lib', {
+    malloc = shared.path_from_root('system', 'lib', {
       'dlmalloc': 'dlmalloc.c', 'emmalloc': 'emmalloc.cpp'
-    }[self.malloc])]
+    }[self.malloc])
+    sbrk = shared.path_from_root('system', 'lib', 'sbrk.c')
+    return [malloc, sbrk]
```

Comment:
Why isn't this just part of libc?  Why compile it N times into each libmalloc if its the same code in each one?

Reply:
I guess it could be part of libc too.

In general there aren't N builds, though, there is normally just one build of malloc. Unless the user builds with flags that actually require a different sbrk build, like pthreads or errno etc. So it wouldn't be the same code in each one.